cql3/statements/modification_statement.cc

```diff
@@ -258,6 +259,10 @@ static thread_local inheriting_concrete_execution_stage<
 
 future<::shared_ptr<cql_transport::messages::result_message>>
 modification_statement::execute(service::storage_proxy& proxy, service::query_state& qs, const query_options& options) const {
+    if (attrs->is_timestamp_set()) {
+        cql3::util::validate_timestamp(get_timestamp(0, options));
+    }
```

Comment:
I would put the if inside the validate_timestamp function. Just take attrs and options as argument of validate_timestamp.

Reply:
get_timestamp calls -> _attrs->get_timestamp(0, options)

Reply:
done